src/vm/jitinterface.cpp

```diff
@@ -2292,27 +2292,31 @@ bool CEEInfo::getSystemVAmd64PassStructInRegisterDescriptor(
         }
         _ASSERTE(methodTablePtr != nullptr);
 
-        // If we have full support for FEATURE_UNIX_AMD64_STRUCT_PASSING, and not just the interface,
+        // If we have full support for UNIX_AMD64_ABI, and not just the interface,
         // then we've cached whether this is a reg passed struct in the MethodTable, computed during
         // MethodTable construction. Otherwise, we are just building in the interface, and we haven't
         // computed or cached anything, so we need to compute it now.
-#if defined(FEATURE_UNIX_AMD64_STRUCT_PASSING)
+#if defined(UNIX_AMD64_ABI)
         bool canPassInRegisters = useNativeLayout ? methodTablePtr->GetLayoutInfo()->IsNativeStructPassedInRegisters()
                                                   : methodTablePtr->IsRegPassedStruct();
-#else // !defined(FEATURE_UNIX_AMD64_STRUCT_PASSING)
+#else // !defined(UNIX_AMD64_ABI)
+        bool canPassInRegisters = false;
         SystemVStructRegisterPassingHelper helper((unsigned int)th.GetSize());
-        bool canPassInRegisters = methodTablePtr->ClassifyEightBytes(&helper, 0, 0, useNativeLayout);
-#endif // !defined(FEATURE_UNIX_AMD64_STRUCT_PASSING)
+        if (th.GetSize() <= CLR_SYSTEMV_MAX_STRUCT_BYTES_TO_PASS_IN_REGISTERS)
```

Comment:
This check is needed to avoid getting this assert:
```
Assert failure(PID 15672 [0x00003d38], Thread: 9976 [0x26f8]): helperPtr->fieldOffsets[i] < CLR_SYSTEMV_MAX_BYTES_TO_PASS_IN_REGISTERS

CORECLR! MethodTable::AssignClassifiedEightByteTypes + 0xDE (0x00007ff9`5d8915be)
CORECLR! MethodTable::ClassifyEightBytesWithManagedLayout + 0xA49 (0x00007ff9`5d897dfd)
CORECLR! CEEInfo::getSystemVAmd64PassStructInRegisterDescriptor + 0x393 (0x00007ff9`5d653a63)
LINUXNONJIT! Compiler::eeGetSystemVAmd64PassStructInRegisterDescriptor + 0x47 (0x00007ff9`7022485f)
LINUXNONJIT! Compiler::fgMorphArgs + 0xA92 (0x00007ff9`702e8f16)
LINUXNONJIT! Compiler::fgMorphCall + 0xC7A (0x00007ff9`702ec412)
LINUXNONJIT! Compiler::fgMorphTree + 0x8AF (0x00007ff9`702fa44f)
LINUXNONJIT! Compiler::fgMorphSmpOp + 0xFAA (0x00007ff9`702f4a96)
LINUXNONJIT! Compiler::fgMorphTree + 0x3D4 (0x00007ff9`702f9f74)
LINUXNONJIT! Compiler::fgMorphStmts + 0x174 (0x00007ff9`702f8708)
    File: d:\git3\coreclr\src\vm\methodtable.cpp Line: 3052
    Image: D:\git3\coreclr\bin\tests\Windows_NT.x64.Checked\Tests\Core_Root\CoreRun.exe
```